src/Core/Managed/Shared/Extensibility/Implementation/TaskTimer.cs

```diff
@@ -62,34 +72,50 @@ public bool IsStarted
         /// <param name="elapsed">The task to run.</param>
         public void Start(Func<Task> elapsed)
         {
-            var newTokenSource = new CancellationTokenSource();
+            DelayedWork delayedWork = new DelayedWork(elapsed);
+            DelayedWork previousWork = Interlocked.Exchange(ref this.latestWork, delayedWork);
+
+            if (previousWork != null)
+            {
+                previousWork.IsCancellationRequested = true;
```

Comment:
In an application that tracks a large volume of telemetry, `TelemetryBuffer` may `Start` and `Cancel` its `TaskTimer` a lot. By replacing the `CancellationTokenSource` with `DelayedWork.IsCancellationRequested` we may be significantly increasing the number of `Delay` tasks that will be allowed to continue running to completion instead of being cancelled. How does the new implementation affect memory allocation?